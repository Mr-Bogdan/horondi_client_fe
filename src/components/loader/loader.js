import React from 'react';
import { useStyles } from './loader.styles';

export const Loader = (props) => {
  const styles = useStyles(props);

  return (
    <div className={styles.wrapper}>
      <div className={styles.ldsDualRing} />
    </div>
  );
};
