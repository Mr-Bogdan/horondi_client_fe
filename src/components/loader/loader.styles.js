import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
  wrapper: {
    width: '100vw',
    height: '100vh',
    overflow: 'hidden',
    display: 'flex',
    animationName: '',
    animationDuration: '',
    gridColumn: (props) => props.gridColumn || 'none'
  },
  ldsDualRing: {
    display: 'inline-block',
    margin: '0 auto',
    alignItems: 'center',
    alignSelf: 'center',
    '&::after': {
      content: '" "',
      display: 'block',
      borderWidth: (props) => (props.width ? parseInt(props.width) / 10 : 6),
      width: (props) => (props.width ? props.width : '64px'),
      height: (props) => (props.height ? props.height : '64px'),
      borderRadius: '50%',
      borderStyle: 'solid',
      borderColor: `${theme.palette.textColor || 'black'} transparent ${
        theme.palette.textColor || 'black'
      } transparent`,
      animation: '$ldsDualRing 1.2s linear infinite'
    }
  },
  '@keyframes ldsDualRing': {
    '0%': {
      transform: 'rotate(0deg)'
    },
    '100%': {
      transform: 'rotate(360deg)'
    }
  }
}));
