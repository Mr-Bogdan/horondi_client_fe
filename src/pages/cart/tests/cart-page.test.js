import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as reactRedux from 'react-redux';
import { Loader } from '../../../components/loader/loader';

import CartPage from '../index';

configure({ adapter: new Adapter() });

let wrapper;

function spyOnSelector(list, loading) {
  jest.spyOn(reactRedux, 'useSelector').mockImplementation((cb) => cb({ Cart: { list, loading } }));
}

describe('Cart page tests', () => {
  it('should match snapshot', () => {
    spyOnSelector([], false);

    wrapper = shallow(<CartPage />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should find loader', () => {
    spyOnSelector([], true);

    wrapper = shallow(<CartPage />);

    expect(wrapper.exists(Loader)).toBe(true);
  });

  it('should cover other branches', () => {
    spyOnSelector([{}], false);

    wrapper = shallow(<CartPage />);
  });
});
