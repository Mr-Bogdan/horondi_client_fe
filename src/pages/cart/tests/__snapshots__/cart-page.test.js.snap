// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart page tests should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <EmptyCart />
</div>
`;
